lib/SILGen/SILGenLValue.cpp

```diff
@@ -4564,10 +4564,19 @@ LValue SILGenLValue::visitABISafeConversionExpr(ABISafeConversionExpr *e,
                                     LValueOptions options) {
   LValue lval = visitRec(e->getSubExpr(), accessKind, options);
   auto typeData = getValueTypeData(SGF, accessKind, e);
+  auto subExprType = e->getSubExpr()->getType()->getRValueType();
+  auto loweredSubExprType = SGF.getLoweredType(subExprType);
```

Comment:
For my understanding: is this equivalent to `lval.getTypeOfRValue()`? I was thinking that instead of having query expression for `subExprType` we could also use `lval.getSubstFormalType()` as well?

Reply:
`lval.getTypeOfRValue()` might be different if the member referenced by the lvalue has a different natural abstraction level. This happens in the test case where you have (simplified)

```
struct Foo<T> {
  var member: () -> T
}

extension Foo where T == ConcreteType {
  func bar() { self.member = ... }
}
```

and you refer to `foo.member`, the storage for `member` uses the generic representation, but in the extension, the type is made concrete so the substituted type of `member` is nongeneric. So `lval.getTypeOfRValue()` here would give the generic representation, whereas `getLoweredType(subExprType)` gives the concrete representation that's expected by the `UncheckedConversionComponent`.

Reply:
Thanks, that was my problem!